core/corehttp/gateway_test.go

```diff
@@ -114,21 +130,27 @@ func TestGatewayGet(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	ns["/ipns/example.com"] = path.FromString("/ipfs/" + k)
+	kTTL := 10 * time.Minute
+	ns["/ipns/example.com"] = mockEntry{
+		path: path.FromString("/ipfs/" + k),
+		ttl:  kTTL,
+	}
 
 	t.Log(ts.URL)
 	for _, test := range []struct {
 		host   string
 		path   string
 		status int
+		ttl    []time.Duration // an approximation for a Maybe/Option sum type
```

Comment:
the clean way around that is to make `namesys.ImmutableTTL` into a `var` instead of a `const`